.editorconfig

```diff
@@ -258,6 +258,8 @@ dotnet_diagnostic.IDE0055.severity = default
 # IDE0061: Use block body for local functions
 dotnet_diagnostic.IDE0061.severity = default
 
+dotnet_diagnostic.BuildErrorsAnalyzer.severity = error
```

Comment:
what is the default severity level? If it is warning, we probably don't need this line.

Reply:
default is error. With this line `dotnet_analyzer_diagnostic.severity = warning`, it will turn into warning. But this will not break `dotnet build` or DevOps pipeline. 
I ever tried `dotnet build --no-incremental -warnAsError .\docfx.csproj`. It will output error but the CLI is still successed: it return 0 and will not break pipeline(continue to execute next command or step).

Reply:
The `-warnAsError` is a known issue with the current dotnet core SDK: https://github.com/dotnet/roslyn/issues/43051 which we could address separately. Consider remove this line from this PR.

Reply:
I removed it and ran pipeline again. From [this log](https://dev.azure.com/ceapex/Engineering/_build/results?buildId=349671&view=logs&j=2b859c7d-aaae-547b-0354-c90791ad9984&t=1beb5644-7360-5729-b1db-6db8eb6a2b48), you can see `dotnet build` output an analyzer warning but  pipeline still continue and succeeded.

Reply:
#7123 should fix this.

Reply:
ya, [the log](https://dev.azure.com/ceapex/Engineering/_build/results?buildId=351377&view=logs&j=a29a2371-0793-5e14-39b9-20eb0beef0db&t=5bb30a61-c77c-5d37-d9a3-6055122d4feb) shows it works - analyzer breaked pipeline